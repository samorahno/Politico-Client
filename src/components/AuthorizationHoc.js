import React from 'react';
import { Route, Redirect } from 'react-router-dom';
import { connect } from 'react-redux';
import axios from '../config/axiosConfig';


const AuthorizationHOC = ({ component: Component, ...rest }) => {
    const token = rest.auth ? rest.auth.token : null;
    const isAdmin = rest.auth.user ? rest.auth.user.isAdmin : null;
    return (
        <Route
            {...rest}
            render={() => {
                if (token) {
                    axios.defaults.headers.common['x-access-token'] = token;
                    return <Component isAdmin={isAdmin}/>;
                }
                return <Redirect to="/" />;
            }}
        />);
};

const mapStateToProps = (state) => ({
    auth: state.auth
});
export default connect(mapStateToProps)(AuthorizationHOC);
