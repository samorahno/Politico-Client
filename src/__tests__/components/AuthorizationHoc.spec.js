import React from 'react';
import { shallow } from 'enzyme';

import { AuthorizationHOC } from '../../components/AuthorizationHoc';

const auth = {
    token: 'fyjfjyhfhjfjhfjy',
    user: {
        isAdmin: true
    }
};

describe('<AuthorizationHOC />', () => {
    let wrapper;

    it('should be a route component', () => {
        wrapper = shallow(<AuthorizationHOC auth={auth} component={() => <div>Hello </div>} />);
        expect(wrapper).toMatchSnapshot();
    });
});
