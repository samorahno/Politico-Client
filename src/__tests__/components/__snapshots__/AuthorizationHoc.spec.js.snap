// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AuthorizationHOC /> should be a route component 1`] = `ShallowWrapper {}`;
